framework/tests/flow2/Purity.java

```diff
@@ -26,6 +28,19 @@ public PureClass() {}
         public SEClass() {}
     }
 
+    // sample test checking for deterministic
+    @MultipleRunDeterministic
+    void mdet() {}
+
+    @SingleRunDeterministic
+    void sdet() {}
+
+    @SingleRunDeterministic
+    void bad() {
+        //:: error: (purity.not.singlerundeterminisitic.call)
+        mdet();
```

Comment:
Isn't SingleRunDeterministic weaker than MultiRunDeterministic? Shouldn't it be safe to call a MRD method from within a SRD method?